src/Build/Instance/ProjectItemInstance.cs

```diff
@@ -1076,6 +1086,28 @@ internal int DirectMetadataCount
                 }
             }
 
+            /// <summary>
+            /// Used to return metadata from another AppDomain. Can't use yield return because the
+            /// generated state machine is not marked as [Serializable], so we need to allocate.
+            /// </summary>
+            /// <param name="list">The source list to return metadata from.</param>
+            /// <returns>An array of string key-value pairs representing metadata.</returns>
+            private IEnumerable<KeyValuePair<string, string>> EnumerateMetadataEager(CopyOnWritePropertyDictionary<ProjectMetadataInstance> list)
+            {
+                var result = new List<KeyValuePair<string, string>>(list.Count);
+
+                foreach (var projectMetadataInstance in list)
+                {
+                    if (projectMetadataInstance != null)
```

Comment:
In what situations is this null?

Reply:
I don't exactly know, if this is never null it could significantly simplify things as then we'd know the Count in advance, but the null check was in the original code and I was extremely careful to preserve the semantics. Certainly not going to risk removing the null check in this release ;) Perhaps in 17.